### ISSUE TITLE
Invalid cast: string[] -> UUID[]

### ISSUE THREAD

**Issue by 0xbaadf00d** [cockroachdb/cockroach#23341]
I'm using an ORM (ebean) and I detect an issue when it try to load many to one / many to many objects. It seems that the cast TEXT[] --> UUID[] not working. The same query works on PostgreSQL 9.6.

```
pq: invalid cast: string[] -> UUID[]
```

**Version**
```
Build Tag:    v1.1.5
Build Time:   2018/02/05 17:44:17
Distribution: CCL
Platform:     windows amd64
Go Version:   go1.8.3
C Compiler:   gcc 6.3.0
Build SHA-1:  ec52ded95a5235d71f26b6cbc59924bd6733fb3a
Build Type:   release
```

**Query**
```
SELECT * FROM identity_provider_cfg WHERE identity_provider_id = any('{4d6c4357-0c43-49ab-84e1-d6d44462b055}'::text[]::uuid[]);
pq: invalid cast: string[] -> UUID[]
```

**Comment by justinj** [comment]
@abhimadan your change should fix this in 2.0, right?

**Comment by abhimadan** [comment]
Yeah, I believe #22338 fixes this issue. If I run `select '{4d6c4357-0c43-49ab-84e1-d6d44462b055}'::text[]::uuid[]` on a recent cockroach binary, I get the following output:
```
+----------------------------------------------------------+
| '{4d6c4357-0c43-49ab-84e1-d6d44462b055}'::TEXT[]::UUID[] |
+----------------------------------------------------------+
| {4d6c4357-0c43-49ab-84e1-d6d44462b055}                   |
+----------------------------------------------------------+
```

**Comment by justinj** [comment]
Thanks @abhimadan! @0xbaadf00d, thanks for the report, this should be resolved in v2.0!
